File: Seminar/src/main/java/org/sopt/Seminar/domain/member/dto/MemberGetResponse.java
package org.sopt.Seminar.domain.member.dto;

import org.sopt.Seminar.domain.member.model.Member;
import org.sopt.Seminar.domain.member.model.SOPT;

public record MemberGetResponse(
		String name,
		String nickname,
		int age,
		SOPT soptInfo
) {
	public static MemberGetResponse of(Member member) {
		return new MemberGetResponse(
				member.getName(),
				member.getNickname(),
				member.getAge(),
				member.getSopt()
		);
	}
}

File: kopring/week2/kotlin-semina/src/main/kotlin/org/sopt/kotlinsemina/domain/member/dto/MemberGetResponse.kt
package org.sopt.kotlinsemina.domain.member.dto

import org.sopt.kotlinsemina.domain.member.model.Member
import org.sopt.kotlinsemina.domain.member.model.SOPT

data class MemberGetResponse(
    val name: String,
    val nickname: String,
    val age: Int,
    val soptInfo: SOPT, // 프로퍼티 private으로 설정 시 jackson 라이브러리가 json에 값 주입을 못해줌.
) {
    companion object {
        fun of(member: Member): MemberGetResponse {
            return MemberGetResponse(
                name = member.name,
                nickname = member.nickname,
                age = member.age,
                soptInfo = member.sopt
            )
        }
    }
}
